core/eth/contracts.go

```diff
@@ -86,9 +101,158 @@ func MustGetV6ContractEventID(name, eventName string) common.Hash {
 		logger.Panic(fmt.Errorf("unable to find contract %s", name))
 	}
 
-	event, found := contract.ABI.Events[eventName]
+	event, found := contract.ABI().Events[eventName]
 	if !found {
 		logger.Panic(fmt.Errorf("unable to find event %s for contract %s", eventName, name))
 	}
 	return event.ID()
 }
+
+func (contract *contract) UnpackLog(out interface{}, event string, log Log) error {
```

Comment:
**Review notes:** `UnpackLog` and `parseTopics` have been copied verbatim from the Geth codebase, which does not expose them in a way we can utilize.